### PR TITLE
fix (PROJ-3746): Add prependLabel option

### PR DESCRIPTION
**Pull request by ewalters-politico**
image

### CODE REVIEW

File: src/SelectSingle.vue [hunk 1]
```diff
@@ -293,7 +300,7 @@
 				<span v-else-if="isPlaceholderShown" class="combo-placeholder">{{ placeholder }}</span>
 				<!-- @slot Display the currently selected option via custom template code -->
 				<slot v-else name="selectedOption" :option="value">
-					{{ value[labelField] }}
+					{{ this.prependLabel ? label + ': ' + value[labelField] : value[labelField] }}
```
Comment (rbvea) [inline src/SelectSingle.vue:303]:
```suggestion
					{{ prependLabel ? label + ': ' + value[labelField] : value[labelField] }}
```

`this` isn't exposed to the template in Vue. Templates have access to the `computed`, `props` and `data` properties without using `this`.

**Review by rbvea — APPROVED** [review]
Ran the storybook locally and I'm seeing an error with this line but if I remove the `this` reference it works.

File: specs/SelectSingle.spec.ts [hunk 2]
```diff
@@ -52,4 +52,68 @@ describe('SelectSingle', () => {
...
+				{ label: 'Item One', value: 'one' },
+				{ label: 'Item Two', value: 'two' },
+				{ label: 'Item Three', value: 'three' }
+			]
+			it('prepends the field label to the displayed option when attribute is set to true', async () => {
+				const myWrapper = mount({
+					data() { 
+						return { selectedOption: {}, options }},
+					template: `<div>
+									<SelectSingle :options="options" v-model="selectedOption" label="Sample SelectSingle" prependLabel="true"/>
```
Comment (rbvea) [inline specs/SelectSingle.spec.ts:68]:
```suggestion
									<SelectSingle :options="options" v-model="selectedOption" label="Sample SelectSingle" :prependLabel="true"/>
```

Also fyi I see this warning when I run `yarn test`

`[Vue warn]: Invalid prop: type check failed for prop "prependLabel". Expected Boolean, got String with value "true".`

File: specs/SelectSingle.spec.ts [hunk 3]
```diff
@@ -52,4 +52,68 @@ describe('SelectSingle', () => {
...
+				await listbox.trigger('click')
+				await listbox.findAll('div[role=option]').at(1).trigger('click')
+				expect(myWrapper.find('div[role=combobox]').text()).toContain('Sample SelectSingle: Item Two')
+			})
+			it('does not prepend the label to items in the list while the dropdown is open', async () => {
+				const myWrapper = mount({
+					data() { 
+						return { selectedOption: {}, options }},
+					template: `<div>
+									<SelectSingle :options="options" v-model="selectedOption" label="Sample SelectSingle" prependLabel="true"/>
```
Comment (rbvea) [inline specs/SelectSingle.spec.ts:82]:
```suggestion
									<SelectSingle :options="options" v-model="selectedOption" label="Sample SelectSingle" :prependLabel="true"/>
```

**Review by rbvea — COMMENTED** [review]
Other warnings about passing in a string instead of using `v-bind` to pass in a boolean.

File: src/SelectSingle.vue (hunk 1 above)
Comment (ewalters-politico) [reply src/SelectSingle.vue]: Fixed, thank you!

File: specs/SelectSingle.spec.ts (hunk 3 above)
Comment (ewalters-politico) [reply specs/SelectSingle.spec.ts]: Fixed, thank you!

File: src/SelectSingle.vue (hunk 1 above)
Comment (ewalters-politico) [reply src/SelectSingle.vue]: Weirdly enough, I was not seeing any errors with my local storybook, I'm not sure why.

**Pull request merged**